### PR TITLE
Added -d flag to migrate_agents. It allows to run agent modification …

### PR DESCRIPTION
**Pull request by konradxyz**
…operation on specific deployment.

### CODE REVIEW

File: common_agents/run.py [hunk 1]
```diff
@@ -55,26 +56,41 @@ def _perform_deployment_updates(deployment_id, updates, update_key, sm):
...
-            deployment.blueprint_id,
-            deployment.id,
-            manager_venv,
-            operation
-        ))
-    finally:
-        _perform_deployment_updates(deployment.id, actions, 'revert', sm)
-    if ret:
-        sys.exit(ret)
+    deployments = sm.deployments_list()
```
Comment (simon-bar) [inline common_agents/run.py]: Please place this line in the else: branch of the IF that follows it

File: common_agents/run.py [hunk 2]
```diff
@@ -55,26 +56,41 @@ def _perform_deployment_updates(deployment_id, updates, update_key, sm):
     map(lambda u: _perform_node_update(u, deployment_id, sm), updates_list)
 
 
-sm = storage_manager_instance()
+def main(args):
+    sm = storage_manager_instance()
 
-manager_venv = sys.argv[1]
-operation = sys.argv[2]
+    manager_venv = sys.argv[1]
```
Comment (simon-bar) [inline common_agents/run.py]: You still use sys.argv here instead of args

File: migrate_agents.sh [hunk 3]
```diff
@@ -26,11 +28,30 @@ function prepare_agents_script {
     sed -i s@__MANAGER_ENV__@$1@ run.sh
     sed -i s@__OPERATION__@$2@ run.sh
     cp $3 auth_config.yaml
+    rm -f deployment_id
```
Comment (simon-bar) [inline migrate_agents.sh]: Is this line necessary?

echo -n $5 > deployment_id    will create/clear the file and fill it thereafter

File: README.md [hunk 4]
```diff
@@ -61,7 +61,13 @@ Utility scripts:
 
 This script is responsible for updating Cloudify components on host-agents machines. These components are Celery workers, plugins code and the monitoring tool - Diamond.
 
-Parameters:
+Parameters and flags:
+
+    -d
+        Usage: -d deployment_id
+        Optional flag. If specified, migrate_agents will perform operation only on hosts
+        that are part of a deployment with id equal to 'deployment_id'.
```
Comment (simon-bar) [inline README.md]: with id equal to  ->  identified by

File: migrate_agents.sh [hunk 5]
```diff
@@ -11,14 +11,16 @@ function cleanup {
...
+    error "Usage: $SCRIPT_NAME [-d deployment_id] (install|uninstall) (3.1|3.2) cli_venv cli_dir [auth_config]" 1
 }
 
 #$1 - manager virtualenv path - it depends on version
 #$2 - operation - either migration_install or migration_uninstall
 #$3 - path to custom authentication configuration
 #$4 - result script path
+#$5 - optional - deployment id
 function prepare_agents_script {
+    rm -rf /tmp/agents_script
```
Comment (simon-bar) [inline migrate_agents.sh]: There is exactly the same line at the end of this function. Maybe we should simply add it to the cleanup() ?

File: migrate_agents.sh [hunk 6]
```diff
@@ -26,11 +28,30 @@ function prepare_agents_script {
...
+    case $opt in
+        d)
+            DEPLOYMENT_ID=$OPTARG
+            ;;
+        \?)
+            usage_exit
+            ;;
+    esac
+done
+shift $((OPTIND - 1))
```
Comment (simon-bar) [inline migrate_agents.sh]: shift influences $# .. is the next line ok now?

File: migrate_agents.sh [hunk 7]
```diff
@@ -73,7 +94,7 @@ else
 fi
 
 echo "Preparing operation script"
```
Comment (simon-bar) [inline migrate_agents.sh]: I would move this line inside the function

File: common_agents/run.py (hunk 2 above)
Comment (konradxyz) [inline common_agents/run.py]: I can only LoL at my stupidity.

File: migrate_agents.sh [hunk 8]
```diff
@@ -6,31 +6,49 @@ BASE_DIR=$(dirname $(readlink -e $0))
...
 #$1 - manager virtualenv path - it depends on version
 #$2 - operation - either migration_install or migration_uninstall
 #$3 - path to custom authentication configuration
 #$4 - result script path
+#$5 - optional - deployment id
 function prepare_agents_script {
-    mkdir -p /tmp/agents_script
-    cp $BASE_DIR/common_agents/* /tmp/agents_script
-    cd /tmp/agents_script
+    TMP_DIR=$(mktemp -d)
```
Comment (simon-bar) [inline migrate_agents.sh]: Some migration related name prefix could be added here, like  mktemp -d /tmp/common_agents_XXXXX

**Pull request merged**